pyface/action/tests/test_traitsui_widget_action.py

```diff
@@ -70,19 +119,40 @@ class ComplexEnumAction(TraitsUIWidgetAction):
 
         model = self.create_model()
         action = ComplexEnumAction(name="Simple", model=model)
-        parent = None
-        control = action.create_control(parent)
+        control = action.create_control(self.parent.control)
+        self.gui.process_events()
 
         editor = control._ui.get_editors('value')[0]
 
         with self.assertTraitChanges(model, 'value', count=1):
-            editor.control.setCurrentIndex(1)
+            if toolkit.toolkit in {'qt', 'qt4'}:
+                editor.control.setCurrentIndex(1)
+                editor.control.activated.emit(1)
+            elif toolkit.toolkit == 'wx':
+                import wx
+                event = wx.CommandEvent(wx.EVT_CHOICE.typeId,
+                                        editor.control.GetId())
+                event.SetString('b')
+                wx.PostEvent(editor.control.GetEventHandler(), event)
+            else:
+                self.skipTest("Unknown toolkit")
+            self.gui.process_events()
 
         self.assertEqual(model.value, 'b')
 
         editor = control._ui.get_editors('value')[1]
 
         with self.assertTraitChanges(action, 'value', count=1):
-            editor.control.setCurrentIndex(2)
+            if toolkit.toolkit in {'qt', 'qt4'}:
+                editor.control.setCurrentIndex(2)
+                editor.control.activated.emit(2)
+            elif toolkit.toolkit == 'wx':
+                event = wx.CommandEvent(wx.EVT_CHOICE.typeId,
+                                        editor.control.GetId())
+                event.SetString('c')
+                wx.PostEvent(editor.control.GetEventHandler(), event)
+            else:
+                self.skipTest("Unknown toolkit")
+            GUI.process_events()
```

Comment:
Why not `self.gui.process_events()`?

Reply:
Because I wrote this line before I added `self.gui`.  Will fix.